src/main/java/de/tum/in/www1/artemis/service/connectors/bitbucket/BitbucketService.java

```diff
@@ -807,6 +816,44 @@ public Commit getLastCommitDetails(Object requestBody) throws BitbucketException
         return commit;
     }
 
+    /**
+     * Retrieves the date at which the push event was received by the VCS intance.
+     *
+     * @param participation The participation we need the date for
+     * @param hash          The hash we expect to find
+     * @return The build queue date
+     */
+    @Override
+    public ZonedDateTime getPushDate(ProgrammingExerciseParticipation participation, String hash) {
+        final var url = bitbucketServerUrl + "/rest/api/latest/projects/" + participation.getProgrammingExercise().getProjectKey() + "/repos/"
+                + urlService.getRepositorySlugFromRepositoryUrl(participation.getVcsRepositoryUrl()) + "/ref-change-activities?start=0&limit=25";
+        final var response = restTemplate.exchange(url, HttpMethod.GET, null, String.class);
+        if (response.getStatusCode() != HttpStatus.OK || response.getBody() == null) {
+            throw new BambooException("Unable to get push date for participation " + participation.getId() + "\n" + response.getBody());
+        }
+        try {
+            // TODO: use a DTO (e.g. something similar to CommitDTO)
+            JsonObject responseJson = (JsonObject) JsonParser.parseString(response.getBody());
+            Assert.notNull(responseJson, "No response");
```

Comment:
please use a proper DTO and avoid `Assert.notNull`
We don't use such assertions in production code and should handle such situations gracefully in my opinion instead of throwing exceptions